src/dex/uniswap-v3/uniswap-v3.ts

```diff
@@ -143,10 +161,23 @@ export class UniswapV3
   ): Promise<UniswapV3EventPool | null> {
     let pool =
       this.eventPools[this.getPoolIdentifier(srcAddress, destAddress, fee)];
+
     if (pool === undefined) {
       const [token0, token1] = this._sortTokens(srcAddress, destAddress);
 
       const key = `${token0}_${token1}_${fee}`.toLowerCase();
+
+      const notExistingPoolScore = await this.dexHelper.cache.zscore(
+        this.notExistingPoolSetKey,
+        key,
+      );
+
+      const poolDoesNotExist = notExistingPoolScore !== null;
+
+      if (poolDoesNotExist) {
+        return null;
```

Comment:
I think before you return `null` here, you need to set:
```
this.eventPools[this.getPoolIdentifier(srcAddress, destAddress, fee)] = pool;
```
So next time it will take locally the value instead of Redis cache

Reply:
@Verisana updated again

Reply:
Thanks, looks good :)